app/uk/gov/hmrc/gform/tasklist/TaskListController.scala

```diff
@@ -154,4 +156,55 @@ class TaskListController(
           )
         ).pure[Future]
     }
+
+  def showDeclaration(
+    maybeAccessCode: Option[AccessCode],
+    formTemplateId: FormTemplateId,
+    coordinates: Coordinates,
+    taskCompleted: Option[Boolean]
+  ): Action[AnyContent] =
+    auth.authAndRetrieveForm[SectionSelectorType.WithDeclaration](
+      formTemplateId,
+      maybeAccessCode,
+      OperationWithForm.ViewDeclaration
+    ) { implicit request => implicit l => cache => implicit sse => formModelOptics =>
+      val maybeDeclarationPage: Option[Singleton[DataExpanded]] =
+        TaskListUtils.withTask(cache.formTemplate, coordinates.taskSectionNumber, coordinates.taskNumber) { task =>
+          for {
+            declarationPage <- task.declarationSection.map(_.toPage)
+          } yield Singleton(declarationPage).asInstanceOf[Singleton[DataExpanded]]
+        }
+      maybeDeclarationPage.fold[Future[Result]](
+        Future.failed(
+          new BadRequestException(
+            s"Declaration Section is not defined in task ${coordinates.taskSectionNumber.value}-${coordinates.taskNumber.value} for ${cache.formTemplateId}"
+          )
+        )
+      ) { declarationPage =>
+        import i18nSupport._
+        for {
+          validationResult <- validationService
+                                .validateAllSections(
+                                  cache.toCacheData,
+                                  formModelOptics.formModelVisibilityOptics,
+                                  EnvelopeWithMapping.empty
+                                )
```

Comment:
Obvious question now is what exactly are we validation here?

I think it would be worth to actually try to write this as a bespoke functionality... ie a new rendering function which doesn't handle any validation result or such.